packages/neuron-wallet/src/services/tx/transaction-persistor.ts

```diff
@@ -157,10 +158,44 @@ export class TransactionPersistor {
       })
     )
 
-    await connection.manager.save([tx, ...inputs, ...previousOutputs, ...outputs])
+    const sliceSize = 100
+    const sleepTime = 10
+    const queryRunner = connection.createQueryRunner()
+    await TransactionPersistor.waitTransaction(queryRunner)
+    await queryRunner.startTransaction()
+    try {
+      await queryRunner.manager.save(tx)
+      for (const slice of Utils.eachSlice(inputs, sliceSize)) {
+        await queryRunner.manager.save(slice)
+        await Utils.sleep(sleepTime)
+      }
+      for (const slice of Utils.eachSlice(previousOutputs, sliceSize)) {
+        await queryRunner.manager.save(slice)
+        await Utils.sleep(sleepTime)
+      }
+      for (const slice of Utils.eachSlice(outputs, sliceSize)) {
+        await queryRunner.manager.save(slice)
+        await Utils.sleep(sleepTime)
+      }
+      await queryRunner.commitTransaction();
+    } catch (err) {
+      await queryRunner.rollbackTransaction()
+    } finally {
+      await queryRunner.release()
+    }
     return tx
   }
 
+  private static waitTransaction = async(queryRunner: QueryRunner, timeout: number = 5000) => {
```

Comment:
What does this mean? Wait for a transaction to do what?

Reply:
Wait transaction end.

Reply:
It seems to be `waitUntilTransactionFinished`

Reply:
updated